test/modules/EdgeHubRestartTester/src/Program.cs

```diff
@@ -0,0 +1,161 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace EdgeHubRestartTester
+{
+    using System;
+    using System.Net;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices;
+    using Microsoft.Azure.Devices.Client;
+    using Microsoft.Azure.Devices.Edge.ModuleUtil;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Extensions.Logging;
+
+    class Program
+    {
+        static readonly ILogger Logger = ModuleUtil.CreateLogger("EdgeHubRestartTester");
+
+        public static int Main() => MainAsync().Result;
+
+        static async Task<int> MainAsync()
+        {
+            uint restartCount = 0;
+
+            Guid batchId = Guid.NewGuid();
+            Logger.LogInformation($"Starting EdgeHubRestartTester ({batchId}) with the following settings:\r\n{Settings.Current}");
+
+            Logger.LogInformation($"EdgeHubRestartTester delay start for {Settings.Current.TestStartDelay}.");
+            await Task.Delay(Settings.Current.TestStartDelay);
+
+            (CancellationTokenSource cts, ManualResetEventSlim completed, Option<object> handler) = ShutdownHandler.Init(Settings.Current.RestartPeriod - TimeSpan.FromMinutes(1), Logger);
```

Comment:
why do you need to change shutdown wait period?

Reply:
This cancellation token is passed down to the thread that sends directMethod/message.  The 5 seconds timeout is not long enough for edgeHub to bootup and the thread to successfully send DM/message.

Reply:
TODO: Investigate the cancellation token is trigger after 5 seconds. ( Does it work for the 2nd restart as well?!? )

Reply:
The timeout was a result from a typo in the deployment file.